pkg/graveler/graveler.go

```diff
@@ -1286,6 +1295,66 @@ func (g *KVGraveler) SaveGarbageCollectionCommits(ctx context.Context, repositor
 	}, err
 }
 
+type GCUncommittedMark struct {
+	BranchID BranchID `json:"branch"`
+	Key      Key      `json:"key"`
+}
+
+func (g *KVGraveler) GetGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, mark GCUncommittedMark) ([]*ValueRecord, *GCUncommittedMark, error) {
+	var entries []*ValueRecord
+	bItr, err := g.ListBranches(ctx, repository)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer bItr.Close()
+	bItr.SeekGE(mark.BranchID)
+
+	count := 0
+	for bItr.Next() {
```

Comment:
You start here a long operation, please make sure that the context is respected by the inner iterators.